tests/testArgumentParser.py

```diff
@@ -289,13 +290,17 @@ def testAtFile(self):
     def testLogDest(self):
         """Test --logdest
         """
-        logFile = "tests_argumentParser_testLog_temp.txt"
-        self.ap.parse_args(
-            config = self.config,
-            args = [DataPath, "--logdest", logFile],
-        )
-        self.assertTrue(os.path.isfile(logFile))
-        os.remove(logFile)
+        logFile = tempfile.NamedTemporaryFile()
+        try:
+            namespace = self.ap.parse_args(
+                config = self.config,
+                args = [DataPath, "--logdest", logFile.name],
+            )
+            namespace.log.info('test logging to the specified file')
+            self.assertTrue(os.path.isfile(logFile.name))
+            self.assertGreater(os.path.getsize(logFile.name), 0)
```

Comment:
This is a better test since it does at least try to show that logging is also working. The problem is that if I update the leak detector as described above to use `lsof` instead of `psutil` then this still does leak a file descriptor. I think this is because the logger is global and I think that is correct because the logger will open the log file and this test code has a file descriptor open on the log file; the test file closes the log file which causes it to be unlinked but the logger still has an open file descriptor (which I think you will be able to demonstrate by writing to the log after the `finally` although I don't know if the logger raises an error if it fails to write the log message). @ktlim any thoughts? Do we use `psutil` on the basis that it's "mostly right"? Do I switch to lsof as it's less portable but seemingly more accurate? Do we fix the logger to allow log destination to be removed?